Symfony/CS/Fixer/Symfony/MethodSeparationFixer.php

```diff
@@ -0,0 +1,254 @@
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Symfony;
+
+use SplFileInfo;
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * Make sure there is one blank line above and below a method.
+ */
+final class MethodSeparationFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(SplFileInfo $file, Tokens $tokens)
+    {
+        $types = array(array(T_CLASS), array(T_INTERFACE));
+        if (defined('T_TRAIT')) {
+            $types[] = array(T_TRAIT);
+        }
+
+        $index = $tokens->getSize() - 1;
+        do {
+            $classIndex = $tokens->getPrevTokenOfKind($index, $types);
+            if (null === $classIndex) {
+                break; // we are done here
+            }
+
+            // figure out where the classy starts
+            $classStart = $tokens->getNextTokenOfKind($classIndex, array('{'));
+
+            // figure out where the classy ends
+            $classEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classStart);
+
+            if ($tokens[$classIndex]->isGivenKind(T_CLASS)) {
+                $this->fixClass($tokens, $classStart, $classEnd);
+            } elseif ($tokens[$classIndex]->isGivenKind(T_INTERFACE)) {
+                $this->fixInterface($tokens, $classStart, $classEnd);
+            } elseif (defined('T_TRAIT') && $tokens[$classIndex]->isGivenKind(T_TRAIT)) {
+                $this->fixTrait($tokens, $classStart, $classEnd);
+            }
+
+            $index = $classIndex - 1;
+        } while (true);
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $classStart
+     * @param int    $classEnd
+     */
+    private function fixClass(Tokens $tokens, $classStart, $classEnd)
+    {
+        $searchMethodFrom = $classEnd;
+        do {
+            $methodIndex = $tokens->getPrevTokenOfKind($searchMethodFrom, array(array(T_FUNCTION)));
+            if (null === $methodIndex || $methodIndex < $classStart) {
+                break; // no (more) methods so nothing left to fix
+            }
+
+            $methodOpen = $tokens->getNextTokenOfKind($methodIndex, array('('));
+
+            // If there is no string (method name) between T_FUNCTION and the opening of the function,
+            // then we are looking at an anonymous function. Don't fix.
+            $anonymous = true;
+            for ($i = $methodIndex; $i < $methodOpen; ++$i) {
+                if (T_STRING === $tokens[$i]->getId()) {
+                    $anonymous = false;
+                    break;
+                }
+            }
```

Comment:
method does not exist on the master version of Tokens
(@keradus is this by design or a merge thing gone wrong somewhere?)